sopel/plugins/rules.py

```diff
@@ -732,58 +738,58 @@ def is_unblockable(self) -> bool:
         """
 
     @abc.abstractmethod
-    def is_user_rate_limited(self, nick: Identifier) -> bool:
+    def is_user_rate_limited(self, nick: Identifier, at_time: Optional[datetime.datetime] = None) -> bool:
         """Tell when the rule reached the ``nick``'s rate limit.
 
-        :return: ``True`` when the rule reached the limit, ``False`` otherwise
+        :param nick: the nick associated with this check.
+        :param at_time: optional time for the rate limit check. If not given, current time will be used.
```

Comment:
A datetime is not a "time". If you say "time", people may wonder if they can use `time.time()`. A `datetime` is a specific type (not to be confused with date + time either).

And, this might sound unimportant, but it actually is: it's an *aware* datetime. Since you can't compare aware datetime with naive datetime, this information matters. 

```suggestion
        :param at_time: optional aware datetime for the rate limit check.
                        If not given, ``utcnow`` will be used.
```